### ISSUE TITLE
Email sent successfully, but SMTPUnknownError prompted

### ISSUE THREAD

**Issue by libook** [mastodon/mastodon#32373]
### Steps to reproduce the problem

1. Use QQ SMTP server.
2. Trigger any sending Email job.

### Expected behaviour

The job is marked as successful and will not be retried.

### Actual behaviour

The email was received, but the job was marked as failed and waiting for a retry.

### Detailed description

The mailbox successfully receives the email, but Mastodon still marks the email task as a failure and retries it multiple times, ultimately resulting in the same email being received multiple times.
Since QQ mailbox is the most widely used mailbox in China, it is hoped that it can be adapted for determining the results of sending jobs.

### Mastodon instance

My own private instance

### Mastodon version

v4.3.0

### Technical details

If this is happening on your own Mastodon server, please fill out those:
- Using docker image: ghcr.io/mastodon/mastodon:v4.3.0

Sidekiq log:
[sidekiq.log](https://github.com/user-attachments/files/17321228/sidekiq.log)

SMTP settings:
SMTP_SERVER=smtp.qq.com
SMTP_PORT=587

**Comment by renchap** [comment]
Unfortunately this report does not contain the SMTP error that happened. I suspect that QQ replied with an unknown SMTP code, that is not supported by our SMTP library, or something similar.

I am unsure how we could get the exact SMTP error / response, maybe @mjankowski has an idea?

**Comment by mjankowski** [comment]
Yeah, based on ` Net::SMTPUnknownError` coming back and a quick glance through the lib, the idea of a successful (but unknown to the lib) response seems possible.

I'm not familiar enough with net/smtp to know if there's increased logging verbosity we could turn on. You could potentially try to recreate the response either by direct connection, or from a rails console, etc - and see what comes back.

Was this working previously and stopped working on 4.3.0 upgrade - or is this a new install?

**Comment by libook** [comment]
> Yeah, based on ` Net::SMTPUnknownError` coming back and a quick glance through the lib, the idea of a successful (but unknown to the lib) response seems possible.
> 
> I'm not familiar enough with net/smtp to know if there's increased logging verbosity we could turn on. You could potentially try to recreate the response either by direct connection, or from a rails console, etc - and see what comes back.
> 
> Was this working previously and stopped working on 4.3.0 upgrade - or is this a new install?

Thanks for the reply.

I don't know about rails developing. I have tried debug it with `swaks`. I hope this helps.
[swaks.log](https://github.com/user-attachments/files/17337131/swaks.log)

swaks returns 6. Refer to its manual:
> 6
> Connection closed unexpectedly. If the close is detected in response to the ' QUIT ' swaks sends following an unexpected response, the error code for that unexpected response is used instead. For instance, if a mail server returns a 550 response to a MAIL FROM: and then immediately closes the connection, swaks detects that the connection is closed, but uses the more specific exit code 23 to detail the nature of the failure. If instead the server return a 250 code and then immediately closes the connection, swaks will use the exit code 6 because there is not a more specific exit code.

**Comment by mjankowski** [comment]
Hmm ... from that log it does look like a successful send from the smtp server perspective (hence the deliveries you are seeing) ... but then from the swaks error code and the behaviour of your app it seems like something wrong is happening in the very last step of connection ending after the send.

Was this working previously and stopped working on 4.3.0 upgrade - or is this a new install?

**Comment by libook** [comment]
> Hmm ... from that log it does look like a successful send from the smtp server perspective (hence the deliveries you are seeing) ... but then from the swaks error code and the behaviour of your app it seems like something wrong is happening in the very last step of connection ending after the send.
> 
> Was this working previously and stopped working on 4.3.0 upgrade - or is this a new install?

It is a new install.
